dpdispatcher/contexts/local_context.py

```diff
@@ -275,7 +278,10 @@ def download(
                         shutil.rmtree(lfile, ignore_errors=True)
                     elif os.path.isfile(lfile) or os.path.islink(lfile):
                         os.remove(lfile)
-                    shutil.copyfile(rfile, lfile)
+                    if not os.path.islink(rfile):
+                        shutil.move(rfile, lfile)
```

Comment:
Consider handling exceptions that may be raised by `shutil.move` and `shutil.copyfile` to ensure that the application does not crash unexpectedly. [important]